nymea.pro

```diff
@@ -1,7 +1,14 @@
 include(nymea.pri)
 
 # Parse and export NYMEA_VERSION_STRING
-NYMEA_VERSION_STRING=$$system('dpkg-parsechangelog | sed -n -e "s/^Version: //p"')
+isEmpty(NYMEA_VERSION) {
+    NYMEA_VERSION_STRING=$$system('dpkg-parsechangelog | sed -n -e "s/^Version: //p"')
+    message("Using dpkg version string $${NYMEA_VERSION_STRING}")
+} else {
+    # qmake NYMEA_VERSION=1.x.x-custom
+    NYMEA_VERSION_STRING="$${NYMEA_VERSION}"
+    message("Using external defined version $${NYMEA_VERSION_STRING}")
+}
```

Comment:
Generally I think this is a good change. A couple of thoughts though:

IMO we should rather remove the dpkg-parsechangelog altogether because it always already produced a warning when building on non debian systems and instead:

a) provide a default for NYMEA_VERSION (e.g. "development")
b) export the env NYMEA_VERSION in the debian/rules.

If I'm not mistaken it should be as simple as `NYMEA_VERSION := $(shell dpkg-parsechangelog -SVersion)` in debian/rules

Reply:
good point, updated